src/EnergyPlus/SizingAnalysisObjects.cc

```diff
@@ -0,0 +1,603 @@
+				ztStepObj[ ztIndex ].subSteps.resize( newNumSubSteps );
+				ztStepObj[ ztIndex ].numSubSteps = newNumSubSteps;
+			}
+		} else {
+			newNumSubSteps = round ( tmpztStepStamp.timeStepDuration / tmpSysStepStamp.TimeStepDuration ) ;
+			ztStepObj[ ztIndex ].subSteps.resize( newNumSubSteps );
+			ztStepObj[ ztIndex ].numSubSteps = newNumSubSteps;
+			ztStepObj[ ztIndex ].hasSystemSubSteps = true;
+		}
+
+
+			// figure out which index this substep needs to go into
+			// the zone step level data are not yet available for minute, but we can get the previous zone step data...
+		lastZnStepIndex = (ztIndex - 1);
+		std::map< int, int >:: iterator end = envrnStartZtStepIndexMap.end();
+		for (std::map< int, int >:: iterator itr = envrnStartZtStepIndexMap.begin(); itr != end; ++itr){
+			if ( itr->second == ztIndex ) { // don't drop back into the previous environment
+					lastZnStepIndex = ztIndex ;
+			}
+		}
+
+		ZoneStepStartMinutes = ztStepObj[lastZnStepIndex].stepEndMinute;
+		if (ZoneStepStartMinutes < 0.0 ) ZoneStepStartMinutes = 0.0;
+
+		tmpSysStepStamp.stStepsIntoZoneStep = round( 
+			(( ( tmpSysStepStamp.CurMinuteStart - ZoneStepStartMinutes ) / MinutesPerHour) 
+			/ tmpSysStepStamp.TimeStepDuration) ) ;
+
+		ztStepObj[ ztIndex ].subSteps[ tmpSysStepStamp.stStepsIntoZoneStep ] = tmpSysStepStamp;
+		ztStepObj[ ztIndex ].subSteps[ tmpSysStepStamp.stStepsIntoZoneStep ].LogDataValue = *p_rVariable;
+
+	}
+
+	void SizingLog::AverageSysTimeSteps( )
+	{
+		Real64 RunningSum;
+
+		for ( auto &Zt : ztStepObj ){
+			if ( Zt.numSubSteps > 0) {
+				RunningSum = 0.0;
+				for ( auto &SysT : Zt.subSteps ) {
+					RunningSum += SysT.LogDataValue;
+				}
+				Zt.logDataValue = RunningSum / double( Zt.numSubSteps );
+			}
+		}
+	}
+
+	void SizingLog::ProcessRunningAverage ()
+	{
+		Real64 RunningSum = 0.0;
+		Real64 divisor = 0.0;
+		divisor = double( timeStepsInAverage );
+
+		std::map< int, int >:: iterator end = ztStepCountByEnvrnMap.end();
+		for (std::map< int, int >:: iterator itr = ztStepCountByEnvrnMap.begin(); itr != end; ++itr){
+//		for (int k = 0; k < NumOfEnvironmentsInLogSet; k++) { // outer loop over environments in log set
+
+//			for (int i = 0; i < ztStepCountByEnvrn[ k ]; i++ ){ // next inner loop over zone timestep steps 
+			for (int i = 0; i < itr->second ; i++ ){ // next inner loop over zone timestep steps 
+
+				if ( timeStepsInAverage > 0 ) {
+					RunningSum = 0.0;
+					for ( int j = 0; j < timeStepsInAverage ; j++ ) { //
+						if ( (i - j) < 0) {
+							RunningSum += ztStepObj[ envrnStartZtStepIndexMap[ itr->first ] ].logDataValue; //just use first value to fill early steps
+						} else {
+							RunningSum += ztStepObj[ ( (i - j) + envrnStartZtStepIndexMap[ itr->first ] ) ].logDataValue;
+						}
+					}
+					ztStepObj[ (i + envrnStartZtStepIndexMap[ itr->first ] ) ].runningAvgDataValue = RunningSum / divisor;
+				}
+			}
+		}
+	}
+
+	ZoneTimestepObject SizingLog::GetLogVariableDataMax( )
```

Comment:
I was about to ask something similar, @EnergyArchmage.  What happens here if none of the values are greater than zero?  Looks like it will return an uninitialized tmpztStepStamp instance.  Is that possible?